src/crimson/os/seastore/seastore_types.cc

```diff
@@ -286,6 +286,11 @@ std::ostream &operator<<(std::ostream &out, data_category_t c)
   }
 }
 
+bool can_inplace_rewrite(extent_types_t type, paddr_t paddr) {
+  return get_extent_category(type) == data_category_t::DATA &&
+    paddr.get_addr_type() == paddr_types_t::RANDOM_BLOCK;
```

Comment:
I think paddr type here is not a condition, it is an invariant.

For example, during replay, if the extent having delta is DATA, its paddr has to belong to RB, and it is the same case during rewrite-extent and RBM-OOL-write when the dirty extent is DATA.

Specifically, can we keep it as simple as `can_inplace_rewrite(type) -> bool`, and if want to validate the paddr type, add asserts in the according logic?

Reply:
Right. Fixed.